### PR TITLE
Feature/currencies

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Adds a command (/currency <amount> <from> in <to>) for converting between currencies. Tested with the test module

### CODE REVIEW

File: modules/currency/kassy.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿{
+  "name": "currency",
+  "version": 2.0,
```
Comment (mrkno) [inline modules/currency/kassy.json]: Wouldn't this technically be 1.0?

Comment (fallaciousreasoning) [inline modules/currency/kassy.json]: Ah yes it would. I just copied that from the other module. Heh. My bad.

File: modules/currency/currency.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+exports.match = function(text, commandPrefix) {
+    return text.startsWith(commandPrefix + 'currency');
+};
+
+exports.help = function() {
+    return [[this.commandPrefix + 'currency <number> <from-currency> in <to-currency>','Converts between currencies.']];
+};
+
+exports.get_exchange = function(callback) {
```
Comment (mrkno) [inline modules/currency/currency.js]: Does this really need to be exported? It should probably be a local

File: modules/currency/currency.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+   });
+}
+
+exports.run = function(api, event) {
+    var query = event.body.substr(10),
+        parts = query.split(' ');
+    
+    //If we don't have the right number of parts, give up
+    if (parts.length !== 4) {
+      api.sendMessage("That looks wrong. You should try harder.", event.thread_id);
```
Comment (mrkno) [inline modules/currency/currency.js]: inconsistent indentation between the var above and here

File: modules/currency/currency.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    this.get_exchange(function (result) {
+      //If we couldn't get the latest data, give up.
+      if (result.error) {
+        api.sendMessage(result.error, event.thread_id);
+        return;
+      }
+
+
```
Comment (mrkno) [inline modules/currency/currency.js]: redundant blank lines

File: modules/currency/currency.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+exports.convert = function(f, to, amount, conversions) {
+  if (!conversions[f]) {
+    return {error: "Unsupported currency '" + f + "'" };
+  }
+
+  if (!conversions[to]) {
+    return {error: "Unsupported currency '" + to + "'" };
+  }
+
+
```
Comment (mrkno) [inline modules/currency/currency.js]: redundant blank lines

File: modules/currency/currency.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      if (result.error) {
+        api.sendMessage(result.error, event.thread_id);
+        return;
+      }
+
+      api.sendMessage("It's about " + result.result + ' ' + parts[3]);
+    });
+};
+
+exports.convert = function(f, to, amount, conversions) {
```
Comment (mrkno) [inline modules/currency/currency.js]: probably should be local, not exported

File: modules/currency/currency.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  if (!conversions[to]) {
+    return {error: "Unsupported currency '" + to + "'" };
+  }
+
+
+  var a = amount/conversions[f] * conversions[to];
+  a = Math.round(a*100) / 100;
+
+  return {result: a }
+}
```
Comment (mrkno) [inline modules/currency/currency.js]: currently code style is
```
return {
        result: a
};```

File: modules/currency/currency.js (hunk 2 above)
Comment (fallaciousreasoning) [inline modules/currency/currency.js]: Fixed. I was blindly copy/pasting from one of our older modules

File: modules/currency/currency.js (hunk 3 above)
Comment (fallaciousreasoning) [inline modules/currency/currency.js]: Fixed. I don't even know what's going on with it. There were 3 space indents :/

File: modules/currency/currency.js (hunk 4 above)
Comment (fallaciousreasoning) [inline modules/currency/currency.js]: Fixed, I think

File: modules/currency/currency.js (hunk 5 above)
Comment (fallaciousreasoning) [inline modules/currency/currency.js]: Fixed

File: modules/currency/currency.js (hunk 6 above)
Comment (fallaciousreasoning) [inline modules/currency/currency.js]: Also fixed

File: modules/currency/currency.js (hunk 7 above)
Comment (fallaciousreasoning) [inline modules/currency/currency.js]: Fixed

**Comment by mrkno** [comment]
Approved

**Comment by Awarua-** [comment]
Any Reason why this is not a kpm module?

**Comment by mrkno** [comment]
Short answer: laziness

Long answer: Cause it seemed simpler just to move this at the same time as all the other modules rather than get another repo created. As with all the others it will become one in the near future so its not really a concern at the moment.

**Pull request merged**